azure-pipelines.yml

```
@@ -142,26 +134,86 @@ stages:
     burgrName: 'qa'
     burgrType: 'qa'
     stageName: 'qa'
-    # .NET analyzers UTs, ITs, build Java and publish SC QG
+    # .NET code analysis, UTs, ITs, build Java and publish SC QG
     stageDisplayName: 'Tests:'
     stageDependencies: build
 
     jobs:
-    - job: runUnitTestsJob
-      displayName: '.NET UTs'
+    - job: runUnitTestsJobNet48
+      displayName: '.NET 4.8 UTs'
+
+      steps:
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download binaries to test'
+        inputs:
+          artifact: TestBinaries
+          targetPath: 'analyzers\tests\SonarAnalyzer.UnitTest\bin'
+
+      - powershell: |
+          . .\scripts\utils.ps1
+
+          CreateFolder("coverage")
+          CreateFolder("TestResults")
+
+          & OpenCover.Console.exe -target:"dotnet.exe" -targetargs:"test analyzers\tests\SonarAnalyzer.UnitTest\bin\$(BuildConfiguration)\net48\SonarAnalyzer.UnitTest.dll --logger trx" -returntargetcode -filter:"+[SonarAnalyzer.CFG]* +[SonarAnalyzer]* +[SonarAnalyzer.CSharp]* +[SonarAnalyzer.Utilities]* +[SonarAnalyzer.VisualBasic]*" -output:coverage/net48.xml -oldStyle
+          Test-ExitCode "ERROR: Unit tests for net48 FAILED."
+        displayName: '.Net UTs'
+
+      - task: PublishPipelineArtifact@1
+        displayName: 'Save coverage files'
+        inputs:
+          path: 'coverage'
+          artifact: DotnetCoverageNet48
+
+      - task: PublishPipelineArtifact@1
+        displayName: 'Save test results files'
+        inputs:
+          path: 'TestResults'
+          artifact: DotnetTestResultsNet48
+
+    - job: runUnitTestsJobNet5
+      displayName: '.NET 5 UTs'
+
+      steps:
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download binaries to test'
+        inputs:
+          artifact: TestBinaries
+          targetPath: 'analyzers\tests\SonarAnalyzer.UnitTest\bin'
+
+      - powershell: |
+          . .\scripts\utils.ps1
+
+          CreateFolder("coverage")
+          CreateFolder("TestResults")
+
+          & OpenCover.Console.exe -target:"dotnet.exe" -targetargs:"test analyzers\tests\SonarAnalyzer.UnitTest\bin\$(BuildConfiguration)\net5.0\SonarAnalyzer.UnitTest.dll --logger trx" -returntargetcode -filter:"+[SonarAnalyzer.CFG]* +[SonarAnalyzer]* +[SonarAnalyzer.CSharp]* +[SonarAnalyzer.Utilities]* +[SonarAnalyzer.VisualBasic]*" -output:coverage/net5.0.xml -oldStyle
+          Test-ExitCode "ERROR: Unit tests for net5.0 FAILED."
+        displayName: '.Net UTs'
+
+      - task: PublishPipelineArtifact@1
+        displayName: 'Save coverage files'
+        inputs:
+          path: 'coverage'
+          artifact: DotnetCoverageNet5
+
+      - task: PublishPipelineArtifact@1
+        displayName: 'Save test results files'
+        inputs:
+          path: 'TestResults'
+          artifact: DotnetTestResultsNet5
 
+    - job: pushQualityGate
+      displayName: '.Net Quality Gate'
```

As it does the whole thing
```suggestion
      displayName: '.Net Analysis'
```